core/src/main/java/tachyon/worker/TachyonWorker.java

```diff
@@ -117,24 +118,31 @@ public static synchronized TachyonWorker createWorker(TachyonConf tachyonConf) {
 
   }
 
-  private static String getMasterLocation(String[] args, TachyonConf conf) {
-    String masterHostname = NetworkUtils.getLocalHostName(conf);
-    int masterPort = conf.getInt(Constants.MASTER_PORT, Constants.DEFAULT_MASTER_PORT);
-    String confFileMasterLoc = masterHostname + ":" + masterPort;
-    String masterLocation;
-    if (args.length < 1) {
-      masterLocation = confFileMasterLoc;
+  private static String setMasterLocation(String masterUrl, TachyonConf conf) {
```

Comment:
it was not used before, this change makes it determine the master location from both TachyonConf and the parameter passed to the main function